pkg/cache/snapshot.go

```diff
@@ -143,9 +143,11 @@ func (c *ClusterQueue) snapshot() *ClusterQueue {
 		Preemption:                    c.Preemption,
 		NamespaceSelector:             c.NamespaceSelector,
 		Status:                        c.Status,
-		AdmissionChecks:               c.AdmissionChecks.Clone(),
-		AdmissionCheckStrategy:        c.AdmissionCheckStrategy,
 	}
+	admissionChecks := make(map[string]sets.Set[string], len(c.AdmissionChecks))
```

Comment:
Wouldn't `maps.Clone` accomplish the same?

Reply:
In this case we needed a deep copy of the set so `maps.Clone` wouldn't be sufficient. I've added a new function that creates a deep copy